src/Cli/dotnet/NugetPackageInstaller/NuGetPackageInstaller.cs

```diff
@@ -107,12 +110,56 @@ public async Task<IEnumerable<string>> ExtractPackageAsync(string packagePath, s
             NuGetPackagePathResolver packagePathResolver = new NuGetPackagePathResolver(targetFolder);
             CancellationToken cancellationToken = CancellationToken.None;
 
-            return await PackageExtractor.ExtractPackageAsync(
+            var allFilesInPackage = await PackageExtractor.ExtractPackageAsync(
                 targetFolder,
                 packageStream,
                 packagePathResolver,
                 packageExtractionContext,
                 cancellationToken);
+
+            if (!OperatingSystem.IsWindows())
+            {
+                foreach (FilePath filePath in FindAllFilesNeedExecutablePermission(allFilesInPackage, targetFolder))
+                {
+                    _filePermissionSetter.Set755Permission(filePath.Value);
+                }
+            }
+
+            return allFilesInPackage;
+        }
+
+        internal IEnumerable<FilePath> FindAllFilesNeedExecutablePermission(IEnumerable<string> files,
+            string targetPath)
+        {
+            if (!PackageIsInAllowList(files))
+            {
+                return Array.Empty<FilePath>();
+            }
+
+            bool FileUnderToolsWithoutSuffix(string p)
+            {
+                return Path.GetRelativePath(targetPath, p).StartsWith("tools" + Path.DirectorySeparatorChar) &&
+                       (Path.GetFileName(p) == Path.GetFileNameWithoutExtension(p));
+            }
+
+            return files
+                .Where(FileUnderToolsWithoutSuffix)
+                .Select(f => new FilePath(f));
+        }
+
+        private static bool PackageIsInAllowList(IEnumerable<string> files)
+        {
+            var allowListOfPackage = new string[] {"microsoft.android.sdk.darwin"};
```

Comment:
@rolfbjarne can confirm tomorrow, but it looks like `Microsoft.iOS.Sdk` also has a `tools` folder where any file without an extension would need execute permission.

Rolf, are there any other files you know of?

Reply:
These are our executable files:



### Microsoft.MacCatalyst.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.MacCatalyst.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      165 Apr 29 10:41:28 2021 Microsoft.MacCatalyst.Sdk/tools/bin/mlaunch*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.MacCatalyst.Sdk/tools/lib/bgen/bgen*

### Microsoft.iOS.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.iOS.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      165 Apr 29 10:41:28 2021 Microsoft.iOS.Sdk/tools/bin/mlaunch*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.iOS.Sdk/tools/lib/bgen/bgen*
-rwxr-xr-x  1 rolf  wheel  6037104 Apr 29 10:41:27 2021 Microsoft.iOS.Sdk/tools/lib/mlaunch/mlaunch.app/Contents/MacOS/mlaunch*

### Microsoft.macOS.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.macOS.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.macOS.Sdk/tools/lib/bgen/bgen*

### Microsoft.tvOS.Sdk:

-rwxr-xr-x  1 rolf  wheel      155 Jan 15 10:08:34 2021 Microsoft.tvOS.Sdk/tools/bin/bgen*
-rwxr-xr-x  1 rolf  wheel      165 Apr 29 10:41:28 2021 Microsoft.tvOS.Sdk/tools/bin/mlaunch*
-rwxr-xr-x  1 rolf  wheel      149 Apr 29 10:36:13 2021 Microsoft.tvOS.Sdk/tools/lib/bgen/bgen*
-rwxr-xr-x  1 rolf  wheel  6037104 Apr 29 10:41:27 2021 Microsoft.tvOS.Sdk/tools/lib/mlaunch/mlaunch.app/Contents/MacOS/mlaunch*

Reply:
So I'll add Microsoft.MacCatalyst.Sdk, Microsoft.iOS.Sdk, Microsoft.macOS.Sdk, Microsoft.tvOS.Sdk

and just mark everything without an extension under tools executable